bug(wizards): Update attributes of Terminal when updating Voltage Level/Bay name.

dlabordus opened the pull request:
closes #674 

- Updated generic reference util to make it possible to search in the document or only in the parent of the element.
For Voltage Level the name is unique inside the Substation and for Bay this is in the Voltage Level.
- Moved some test from zeroline to wizards, because these test are only testing these wizards.
- Renamed "foundation.ts" in test directory to "test-support.ts". There is a more clear differences now between foundation.ts and foundation.test.ts in the test directory.

JakobVogelsang commented:
@dlabordus I have to apologize. I have moved files that are related to this fix and not seeing you are working on the same files. I will fix the missing tests.

However, I did the first review and the implementation looks good to me. 
There is one thing that I have to double-check, though. You are assuming that the `Terminal` element is only within the parent element of the name element you want to change. I am not sure if this always true :)

dlabordus commented:
@JakobVogelsang, already fixed the merge. You just moved some files, but I also moved these files. 
The Test Files where not related to the Editor, but to the Wizards, so I moved these to the Wizards directory.
This way we don't create double test, like what happened with Substation Wizard.

dlabordus commented:
@JakobVogelsang, I updated the reference function to include parent names if needed. This should work for all situation we currently know (, I think :-) )

JakobVogelsang commented on src/wizards/foundation/references.ts:
```diff
@@ -1,54 +1,132 @@
+ * The query will then become
+ * `Terminal[substationName="<substationName>"][voltageLevelName="<voltageLevelName>"][bayName="<oldName>"]`
+ *
+ * @param elementName - The tagName of the elements to search for.
```
The function is working perfectly :) Thanks for that. There is only one remark about naming
`elementName` looking at our code could imply the attribute name of the element

dlabordus replied:
Fixed.

JakobVogelsang commented on src/wizards/foundation/references.ts:
```diff
@@ -1,54 +1,132 @@
+ * The query will then become
+ * `Terminal[substationName="<substationName>"][voltageLevelName="<voltageLevelName>"][bayName="<oldName>"]`
+ *
+ * @param elementName - The tagName of the elements to search for.
```
The function is working perfectly :) Thanks for that. There is only one remark about naming
`elementName` looking at our code could imply the attribute name of the element

dlabordus replied:
Fixed.

JakobVogelsang reviewed (approved):
LGTM aside of one naming issue as a suggestion
